src/main/java/com/ericsson/ei/mongo/MongoUri.java

```diff
@@ -93,24 +99,26 @@ public static String extractPasswordFromUri(String uri) {
             String authenticationDetails = matcher.group(0).replace("//", "");
             if (authenticationDetails.contains(":")) {
                 String[] authenticationDetailsList = authenticationDetails.split(":");
-                secret = authenticationDetailsList[1];
+                password = authenticationDetailsList[1];
             }
         }
-        return secret;
+        return password;
     }
 
-
     /**
-     * By ensuring password is behoind : we know we do not replace username in case username and
-     * password is the same.
+     * The password will always have ':' in front. Username has ':' after. By replacing (':' +
```

Comment:
I don't think you need this explanation in the Javadoc. The user of this API really only needs to know what is being replaced. Not the entire logic behind how it's done.